beta/src/pages/apis/usestate.md

```diff
@@ -87,12 +87,16 @@ function handleClick() {
 
 `set` functions do not have a return value.
 
-#### Caveats {/*caveats*/}
+#### Caveats {/*setstate-caveats*/}
 
 * The `set` function only updates the state variable for the next render. If you read the state variable after calling the `set` function, you will still get the value that was on the screen before your call.
 
 * If the new value you provide is identical to the current `state`, as determined by an [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will bail out of re-rendering the component and its children. However, in some cases React may still need to call your component before discarding the result.
```

Comment:
"However, in some cases React may still need to call your component before discarding the result."

Could you provide an example? Otherwise it may induce a low-key anxiety similar to "I don't know what I don't know."

Also, I'm proposing to change the "bail out" because it's not simple English:

```suggestion
If the new value you provide is identical to the current `state`, as determined by an [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will not re-render the component and its children. However, in some cases React may still need to call your component before discarding the result.
```

Reply:
>Could you provide an example?

This is tricky — *the whole point* is that we're not giving any guarantees and when that might happen, and it *should not matter* for your code. I don't know how to communicate this.

Reply:
How about:

```suggestion
If the new value you provide is identical to the current `state`, as determined by an [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will not re-render the component and its children. Although in some cases React may still need to call your component before discarding the result, that bears no consequence for your code.
```